### PR TITLE
Add model, controller, and tests for getArticleComments

### PR DESCRIPTION
**Pull request by jamalxwm**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -205,3 +205,55 @@ describe('5. GET - /api/articles', () => {
...
+      .get('/api/articles/coconuts/comments')
+      .expect(400)
+      .then(({ body }) => expect(body.msg).toBe('Bad request'));
+  });
+  test('Status: 200 // Returns an array of objects', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then(({ body }) => {
+        const comments = body;
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: data should come back on the key of comments

File: __tests__/app.test.js [hunk 2]
```diff
@@ -205,3 +205,55 @@ describe('5. GET - /api/articles', () => {
...
+        expect(comments.length).toBe(0);
+      });
+  });
+  test('Status: 200 // Comment objects include props for comment_id, votes, created_at, author, and body', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then(({ body }) => {
+        const comments = body;
+        expect(comments.length).toBeGreaterThan(0);
```
Comment (Poonam-raj) [inline __tests__/app.test.js:247]: you should be more precise with this length check, you can look at the test data and see how many comments are associated with this article

File: __tests__/app.test.js [hunk 3]
```diff
@@ -205,3 +205,55 @@ describe('5. GET - /api/articles', () => {
...
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
```
Comment (Poonam-raj) [inline __tests__/app.test.js:255]: would ideally see article_id in the key too

File: controllers/articles.js [hunk 4]
```diff
@@ -1,24 +1,36 @@
...
   updateVotesOnArticleByID(id, vote)
     .then((article) => res.status(200).send(article))
     .catch((err) => next(err));
 };
+
+exports.getArticleComments = (req, res, next) => {
+  const { article_id: id } = req.params;
+  fetchArticleComments(id)
+    .then((comments) => {
+      res.status(200).send(comments);
```
Comment (Poonam-raj) [inline controllers/articles.js]: looking for `.send({comments: comments})` structure

File: models/articles.js [hunk 5]
```diff
@@ -50,10 +51,24 @@ exports.updateVotesOnArticleByID = (id, vote) => {
...
+    const article = await this.fetchArticleByID(id);
+    if (article.status === 404) return Promise.reject(article);
+    const { rows } = await db.query(
+      `SELECT * FROM comments WHERE article_id = $1;`,
+      [id]
+    );
+    return rows;
+  } catch (err) {
+    return Promise.reject(err);
+  }
```
Comment (Poonam-raj) [inline models/articles.js]: if you have a catch block in your controller the error produced by a fail in this model should just be passed down the promise chain into that catch

aka you don't need a try catch block here as your catch in the controller will be passed this error by express

**Review by Poonam-raj — APPROVED** [review]
few amendments needed - happy for you to  proceed once you've looked at my comments and merge yourself 🎉

File: controllers/articles.js (hunk 4 above)
Comment (jamalxwm) [reply controllers/articles.js]: OH MY DAYS, 

This explains why I couldn't put body on the key of comments. I was busting my brain over this for so long!
✅

File: models/articles.js (hunk 5 above)
Comment (jamalxwm) [reply models/articles.js]: Amazing!

**Pull request merged**